### PR TITLE
Fixes a mousetrap exploit involving click+dragging your backpack (or opening it) while stunned to instantly annihilate everyone around you

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: Manly Dwarf
fix: Fixed an exploit related to mousetrap bombs exploding despite the user being stunned or handcuffed.
balance: Also slowed down the explosion a bit so they aren't outright better than voice analyzers.
/:cl:

## About The Pull Request
steamp0rt in shambles

## Why It's Good For The Game
scientists in shambles

destroy science when

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
hahahahahahah

**Comment by DONKARMA** [comment]
Mad?

File: code/modules/assembly/mousetrap.dm [hunk 1]
```diff
@@ -63,6 +63,7 @@
 	playsound(src, 'sound/effects/snap.ogg', 50, TRUE)
 	armed = FALSE
 	update_icon()
+	sleep(1.5) // hippie -- adds a small enough delay to prevent mousetrap signalers from being better voice analyzers
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:66]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 2]
```diff
@@ -116,13 +117,15 @@
 	..()
 
 
-/obj/item/assembly/mousetrap/on_found(mob/finder)
+/obj/item/assembly/mousetrap/on_found(mob/living/finder) // hippie -- consider this whole section edited
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:120]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 3]
```diff
@@ -116,13 +117,15 @@
 	..()
 
 
-/obj/item/assembly/mousetrap/on_found(mob/finder)
+/obj/item/assembly/mousetrap/on_found(mob/living/finder) // hippie -- consider this whole section edited
 	if(armed)
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:121]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 4]
```diff
@@ -116,13 +117,15 @@
...
 
-/obj/item/assembly/mousetrap/on_found(mob/finder)
+/obj/item/assembly/mousetrap/on_found(mob/living/finder) // hippie -- consider this whole section edited
 	if(armed)
-		if(finder)
-			finder.visible_message("<span class='warning'>[finder] accidentally sets off [src], breaking their fingers.</span>", \
-							   "<span class='warning'>You accidentally trigger [src]!</span>")
-			triggered(finder, (finder.active_hand_index % 2 == 0) ? BODY_ZONE_PRECISE_R_HAND : BODY_ZONE_PRECISE_L_HAND)
-			return TRUE	//end the search!
+		if(finder && ishuman(finder))
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:122]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 5]
```diff
@@ -116,13 +117,15 @@
...
+			var/mob/living/carbon/human/H = finder
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:123]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 6]
```diff
@@ -116,13 +117,15 @@
...
+			if(!H.handcuffed && !H.incapacitated())
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:124]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 7]
```diff
@@ -116,13 +117,15 @@
...
+				finder.visible_message("<span class='warning'>[finder] accidentally sets off [src], breaking their fingers.</span>", \
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:125]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 8]
```diff
@@ -116,13 +117,15 @@
...
+								   "<span class='warning'>You accidentally trigger [src]!</span>")
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:126]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 9]
```diff
@@ -116,13 +117,15 @@
...
+				triggered(finder, (finder.active_hand_index % 2 == 0) ? BODY_ZONE_PRECISE_R_HAND : BODY_ZONE_PRECISE_L_HAND)
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:127]: modularize this

File: code/modules/assembly/mousetrap.dm [hunk 10]
```diff
@@ -116,13 +117,15 @@
...
+				return TRUE	//end the search!
```
Comment (ghost) [inline code/modules/assembly/mousetrap.dm:128]: modularize this

**Review by ghost — CHANGES_REQUESTED** [review]
FUCK YOU ALL

**Comment by Manly-Dwarf** [comment]
bling
SCIENTISTS WILL NEVER RECOVER FROM THIS

**Comment by DONKARMA** [comment]
why modularize without a mirror bot

**Comment by Manly-Dwarf** [comment]
karma why are you here

**Comment by DONKARMA** [comment]
i have a fat penis

**Comment by DONKARMA** [comment]
also pillarmen

**Comment by steamp0rt** [comment]
this PR is @DONKARMA's fault btw

**Review by ghost — APPROVED** [review]
sadly me keeping this from being merged is an uphill battle

voice sensors + health sensors are still good so whatever

**Comment by Manly-Dwarf** [comment]
HAHAHAHAHAHAHAH

**Comment by DONKARMA** [comment]
based

**Comment by ghost** [comment]
i'm nuking @DONKARMA from orbit though

**Comment by DONKARMA** [comment]
nope

**Comment by SireTurret** [comment]
amari homosexual, tibby the kiefer wannabe and other science mains btfo btfo btfo

**Comment by Manly-Dwarf** [comment]
peepee

**Comment by ghost** [comment]
this shit is why karma got banned

**Comment by SireTurret** [comment]
makes us all want to bring him back

**Comment by ghost** [comment]
he's too mentally inept to do a ban appeal though

(edited to remove no-no word so github police don't take me to re-education camp)

**Comment by DONKARMA** [comment]
i just don't want to

**Pull request merged**